force-app/main/default/classes/OppClosedWonDeletionValidation.cls

```diff
@@ -0,0 +1,45 @@
+public with sharing class OppClosedWonDeletionValidation {
+    public static void oppsToBeDeleted (List<Opportunity> deletedOpps) {
+        // First build a Set of Account Id's and a set of Opportunity Id's with each Opportunity where the StageName == 'Closed Won'
+        Set<Id> closedWonOppIds = new Set<Id>();
+        for (Opportunity opp : deletedOpps) {
+            if (opp.StageName == 'Closed Won') {
+                closedWonOppIds.add(opp.Id);
+            }
```

Comment:
The only Ids you really need to care about in the problem are the Account Ids. You need to get those to see if they are of Type banking.